File: ClinicaVeterinaria/logica/FlujoTicket.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace ClinicaVeterinaria.logica
{
    public class FlujoTicket
    {
        private Dictionary<string, List<string>> dicTransiciones;

        public FlujoTicket()
        {
            dicTransiciones = new Dictionary<string, List<string>>
            {
                { "Abierto", new List<string> { "Asignado" } },
                { "Asignado", new List<string> { "Escalado", "Resuelto" } },
                { "Escalado", new List<string> { "Asignado", "Resuelto" } },
                { "Resuelto", new List<string> { "Cerrado", "Asignado" } },
                { "Cerrado", new List<string>() }
            };
        }

        public bool puedeCambiarEstado(string strActual, string strNuevo)
        {
            return dicTransiciones.ContainsKey(strActual) && dicTransiciones[strActual].Contains(strNuevo);
        }

        public void mostrarFlujo()
        {
            Console.WriteLine("Flujo: Abierto -> Asignado -> Escalado -> Resuelto -> Cerrado");
            foreach (var par in dicTransiciones)
            {
                string strDestinos = par.Value.Count > 0 ? string.Join(", ", par.Value) : "(estado final)";
                Console.WriteLine($"  {par.Key} -> {strDestinos}");
            }
        }
    }
}

File: BlueJ_Proyecto/FlujoTicket.java
import java.util.ArrayList;
import java.util.HashMap;

/**
 * Valida las transiciones de estado permitidas para un ticket (consulta).
 */
public class FlujoTicket
{
    private HashMap<String, ArrayList<String>> transiciones;

    public FlujoTicket()
    {
        transiciones = new HashMap<>();

        ArrayList<String> desdeAbierto = new ArrayList<>();
        desdeAbierto.add("Asignado");
        transiciones.put("Abierto", desdeAbierto);

        ArrayList<String> desdeAsignado = new ArrayList<>();
        desdeAsignado.add("Escalado");
        desdeAsignado.add("Resuelto");
        transiciones.put("Asignado", desdeAsignado);

        ArrayList<String> desdeEscalado = new ArrayList<>();
        desdeEscalado.add("Asignado");
        desdeEscalado.add("Resuelto");
        transiciones.put("Escalado", desdeEscalado);

        ArrayList<String> desdeResuelto = new ArrayList<>();
        desdeResuelto.add("Cerrado");
        desdeResuelto.add("Asignado");
        transiciones.put("Resuelto", desdeResuelto);

        transiciones.put("Cerrado", new ArrayList<>());
    }

    public boolean puedeCambiarEstado(String estadoActual, String estadoNuevo)
    {
        return transiciones.containsKey(estadoActual) && transiciones.get(estadoActual).contains(estadoNuevo);
    }

    public void mostrarFlujo()
    {
        System.out.println("Flujo: Abierto -> Asignado -> Escalado -> Resuelto -> Cerrado");
        for (String estado : transiciones.keySet())
        {
            ArrayList<String> destinos = transiciones.get(estado);
            String texto = destinos.isEmpty() ? "(estado final)" : String.join(", ", destinos);
            System.out.println("  " + estado + " -> " + texto);
        }
    }
}
